### PR TITLE
[Placement Group]Add get all placement group api

### PR DESCRIPTION
**Pull request by clay4megtr**
## Why are these changes needed?
Add get all placement group api

Next pr will implements java api.

## Related issue number
#9802 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/includes/global_state_accessor.pxi [hunk 1]
```diff
@@ -122,6 +122,12 @@ cdef class GlobalStateAccessor:
             result = self.inner.get().AddWorkerInfo(cserialized_string)
         return result
 
+    def get_placement_group_table(self):
```
Comment (rkooo567) [inline python/ray/includes/global_state_accessor.pxi:125]:
```suggestion
    def get_all_placement_group_info(self):
```

Let's use the consistent term here.

File: python/ray/state.py [hunk 2]
```diff
@@ -388,8 +388,15 @@ def placement_group_table(self, placement_group_id=None):
                                         FromString(placement_group_info))
                 return self._gen_placement_group_info(placement_group_info)
         else:
-            raise NotImplementedError(
-                "Get all placement group is not implemented yet.")
+            placement_group_table = self.global_state_accessor.get_placement_group_table()
+            results = {}
+            for i in range(len(placement_group_table)):
```
Comment (rkooo567) [inline python/ray/state.py:393]:
```suggestion
            for placement_group_info in placement_group_table
```

File: python/ray/tests/test_placement_group.py [hunk 3]
```diff
@@ -415,6 +415,9 @@ def value(self):
     result = ray.util.placement_group_table(placement_group)
     assert result["state"] == "CREATED"
 
+    # Test get all placement group.
```
Comment (rkooo567) [inline python/ray/tests/test_placement_group.py:418]: Can you test it with multiple placement group as well (not just one)?

File: python/ray/util/placement_group.py [hunk 4]
```diff
@@ -242,7 +242,7 @@ def check_placement_group_index(placement_group: PlacementGroup,
             raise ValueError("If placement group is not set, "
                              "the value of bundle index must be -1.")
     elif bundle_index >= placement_group.bundle_count \
-            or bundle_index < -1:
+        or bundle_index < -1:
```
Comment (rkooo567) [inline python/ray/util/placement_group.py:245]: Can you indent here back again? Is it formatted automatically? 

If or is aligning with the line below, it looks confusing.

File: src/ray/gcs/gcs_client/service_based_accessor.cc [hunk 5]
```diff
@@ -1536,5 +1536,20 @@ Status ServiceBasedPlacementGroupInfoAccessor::AsyncGet(
...
 }
 
+Status ServiceBasedPlacementGroupInfoAccessor::AsyncGetAll(
+    const MultiItemCallback<rpc::PlacementGroupTableData> &callback) {
+  RAY_LOG(DEBUG) << "Getting all placement group info.";
+  rpc::GetAllPlacementGroupRequest request;
+  client_impl_->GetGcsRpcClient().GetAllPlacementGroup(
+      request,
+      [callback](const Status &status, const rpc::GetAllPlacementGroupReply &reply) {
+        auto result = VectorFromProtobuf(reply.placement_group_table_data());
```
Comment (rkooo567) [inline src/ray/gcs/gcs_client/service_based_accessor.cc:1546]:
```suggestion

```

File: src/ray/gcs/gcs_client/service_based_accessor.cc [hunk 6]
```diff
@@ -1536,5 +1536,20 @@ Status ServiceBasedPlacementGroupInfoAccessor::AsyncGet(
...
+        callback(status, result);
```
Comment (rkooo567) [inline src/ray/gcs/gcs_client/service_based_accessor.cc:1547]:
```suggestion
        callback(status, VectorFromProtobuf(reply.placement_group_table_data()););
```

**Review by rkooo567 — COMMENTED** [review]
Can you add cpp unit tests?

File: python/ray/tests/test_placement_group.py (hunk 3 above)
Comment (clay4444) [reply python/ray/tests/test_placement_group.py:418]: yes~ I will add more placement group for test.

File: python/ray/tests/test_placement_group.py [hunk 7]
```diff
@@ -374,7 +374,6 @@ def f():
     # Make sure this task is still schedulable.
     assert ray.get(f.remote()) == 3
 
-
```
Comment (ffbin) [inline python/ray/tests/test_placement_group.py:377]: Two blank lines are required between different functions.

File: python/ray/includes/global_state_accessor.pxi (hunk 1 above)
Comment (clay4444) [reply python/ray/includes/global_state_accessor.pxi:125]: hi, @rkooo567 , the other naming methods in this file are all `get_object_info`<->`get_object_info`, `get_actor_table` <-> `get_actor_info`, `get_worker_table`<->`get_worker_info` ...  are we going to change them to be inconsistent (file level)

File: python/ray/state.py (hunk 2 above)
Comment (clay4444) [reply python/ray/state.py]: updated.

File: src/ray/gcs/gcs_client/service_based_accessor.cc (hunk 5 above)
Comment (clay4444) [reply src/ray/gcs/gcs_client/service_based_accessor.cc]: updated.

File: src/ray/gcs/gcs_client/service_based_accessor.cc (hunk 6 above)
Comment (clay4444) [reply src/ray/gcs/gcs_client/service_based_accessor.cc]: updated.

File: python/ray/util/placement_group.py (hunk 4 above)
Comment (clay4444) [reply python/ray/util/placement_group.py:246]: Is this github display problem?  my local display is right.

File: python/ray/tests/test_placement_group.py (hunk 7 above)
Comment (clay4444) [reply python/ray/tests/test_placement_group.py]: updated

**Comment by clay4444** [comment]
cpp unit test added, ( but not add create placement group unti test, because there is no better way to achive this in ut env. )

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**